### PR TITLE
Course controller tests

### PR DESCRIPTION
**Pull request by Devesh21700Kumar**
Terminal commands to test code:-
1).npm run lint:fix
2).nest start
3).npm run test

image

- [x] My code follows the code style of this project.
- [x] All new and existing tests passed.
- [x] My changes generate no new warnings
- [x] The title of my pull request is a short description of the requested changes.

**Screenshots**
   Original                 |   Updated
  :------------------------:|---------------------:
  ** original screenshot ** | ** updated screenshot **

### CODE REVIEW

File: src/modules/course/course.controller.spec.ts [hunk 1]
```diff
@@ -1,25 +1,74 @@
...
+      .mockImplementation((id: string) => ({ ...mockCourse, id })),
+    editCourse: jest.fn().mockImplementation((uid, body) => ({
+      ...mockCourse,
+      id: uid,
+      ...body,
+    })),
+    // deleteCourse: jest.fn().mockResolvedValue([mockCourse]),
+    // getEnrolledCourses: jest.fn().mockResolvedValue([mockCourse]),
+    // addCourse: jest.fn().mockResolvedValue([mockCourse]),
+    // getWishList: jest.fn().mockResolvedValue([mockCourse]),
```
Comment (Abhishek-kumar09) [inline src/modules/course/course.controller.spec.ts:38]: Please add these

Comment (Devesh21700Kumar) [reply src/modules/course/course.controller.spec.ts:38]: Ok I'll convert to draft and add these also

**Comment by Devesh21700Kumar** [comment]
@Abhishek-kumar09 

create course, update course and delete course controller tests have been added


image

File: src/modules/course/course.controller.spec.ts [hunk 2]
```diff
@@ -1,25 +1,148 @@
...
+
+    it('should be found by ID', async () => {
+      const id = new mongoose.Schema.Types.ObjectId('22', 0, 'rtex');
+      await expect(controller.getSelectedCourse(id)).resolves.toEqual({
+        ...mockCourse,
+        id,
+      });
+      expect(service.findCourseById).toHaveBeenCalledWith(id);
+    });
+
```
Comment (Abhishek-kumar09) [inline src/modules/course/course.controller.spec.ts:83]: Also add a case for unavailable Uid and a null uid

Comment (Devesh21700Kumar) [reply src/modules/course/course.controller.spec.ts:83]: ok. sure

**Comment by Devesh21700Kumar** [comment]
@Abhishek-kumar09 added tests to check the case of id which does not exist , so that the value returned is not equal to the expected value

**Review by Abhishek-kumar09 — APPROVED** [review]

**Pull request merged**